src/main/java/org/broadinstitute/hellbender/utils/read/markduplicates/sparkrecords/TransientFieldPhysicalLocation.java

```diff
@@ -0,0 +1,50 @@
+package org.broadinstitute.hellbender.utils.read.markduplicates.sparkrecords;
+
+import picard.sam.util.PhysicalLocation;
+
+/**
+ * A common interface for holding the fields in PhysicalLocation
+ */
+public abstract class TransientFieldPhysicalLocation extends PairedEnds implements PhysicalLocation {
+    // Information used to detect optical dupes
+    protected short readGroupIndex = -1;
+    protected transient short tile = -1;
+    protected transient int x = -1;
+    protected transient int y = -1;
+    protected transient short libraryId = -1;
+
+    public TransientFieldPhysicalLocation(int partitionIndex, String name) {
+        super(partitionIndex, name);
+    }
+
+    // Methods for OpticalDuplicateFinder.PhysicalLocation
+    @Override
+    public short getReadGroup() { return this.readGroupIndex; }
```

Comment:
it's weird that these read group isn't transient but the others are, should you javadoc that?